### PR TITLE
Upgrade of, and adaption to D3 to version 4

### PR DESCRIPTION
**Pull request by magjac**
You may want to have a look at the latest commit, 495567b4d3c3a1afe8ec80374831ecea15bb644b. I really don't understand why it is needed for labels (text), but not for shapes (path). The tests ran fine without it, but the initial transition was very funny, with shapes dropping down linearly from the top, but labels spreading of from the top left corner.

I've done a few manual checks with my local graphviz.it, but I would appreciate if you could also do some.

### CODE REVIEW

**Comment by magjac** [comment]
I'll have a look at the decreased coverage later.

File: src/js/stage.js [hunk 1]
```diff
@@ -213,8 +213,11 @@ define(["d3", "palette", "transitions/default"], function (d3, palette, defaults
           .data(function (d) {
             return d.labels;
           });
-        labels.enter().append("text");
         transitions.labels(labels, labelAttributer);
+        labels.enter()
+          .append("text")
+          .call(labelAttributer)
+          .call(styliseur);
```
Comment (mstefaniuk) [inline src/js/stage.js]: I think that it is not needed only for d3.js version switch but it appears to fix a bug. Am I right?

Comment (mstefaniuk) [inline src/js/stage.js]: Stage should not have dependency to `styliseur`. It should be moved to `default/transitions` if it is really needed. In fact transitions for labels looks as follows:
```js
      labels
        .transition()
        .delay(150)
        .duration(900)
        .call(attributer)
        .call(styliseur);
```
Please make it consistent.

**Comment by mstefaniuk** [comment]
@magjac Forget so small coverage drop. It may be caused by need to add `selection` argument to selection manipulation functions. Please look into my review comment. After fixing that I will merge PR with pleasure.

File: src/js/stage.js (hunk 1 above)
Comment (magjac) [inline src/js/stage.js]: I don't understand your first comment. I don't know if there is a bug. Do you mean in graph-vis-d3-js or in D3? Did you try e.g. http://graphviz.it/#/gallery/mike.gv and compare with my PR without the last commit. I thinks it's not an acceptable behavior.

Regarding your second comment; It seems that it is necessary to set the initial attributes for the enter selection directly before transition of the higher level elements starts. That's why I did it this way and applied the transition to the update selection only. I will investigate if it is possible to apply the transition to a merged selection and get rid of the styliseur dependency.

**Comment by magjac** [comment]
Now a better solution in stage without dependency to styliseur. Please see also my comments to your review regarding possible bug.

**Comment by mstefaniuk** [comment]
Maybe I expected you to fix bug because you introduced new logic in label rendering and most of the changes has been related to new d3 API but not this one.

**Comment by magjac** [comment]
If there is a bug, it was introduced or revealed by the change from D3 v3 to v4. There was no problem before that. You can see for yourself as I explained in my comment to your initial review comments.

If there is a bug, my change is not a fix, but a workaround. But since I don't understand how initial transitions (i.e. for enter selections) work even for the shapes which do not have any workaround, I don't know if it is the correct way or not. That's why I wanted you to review it.

Anyway, thanks for accepting my PR 😃.

**Pull request merged**